webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -1,106 +1,139 @@
 import { Form, Input, message, Switch } from 'antd';
 import { FormInstance } from 'antd/lib/form/hooks/useForm';
-import React, { useCallback } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 
 import { useStore } from 'contexts/Store';
-import { postUser } from 'services/api';
+import { getUser, patchUser, postUser } from 'services/api';
 import useModal, { ModalHooks } from 'shared/hooks/useModal/useModal';
 import { ErrorType } from 'shared/utils/error';
-import { BrandingType } from 'types';
+import { BrandingType, DetailedUser } from 'types';
 import handleError from 'utils/error';
 
-export const MODAL_HEADER_LABEL = 'Create User';
+export const MODAL_HEADER_LABEL_CREATE = 'Create User';
+export const MODAL_HEADER_LABEL_EDIT = 'Edit User';
 export const USER_NAME_NAME = 'username';
 export const USER_NAME_LABEL = 'User Name';
 export const DISPLAY_NAME_NAME = 'displayName';
 export const DISPLAY_NAME_LABEL = 'Display Name';
 export const ADMIN_NAME = 'admin';
 export const ADMIN_LABEL = 'Admin';
-export const API_SUCCESS_MESSAGE = `New user with empty password has been created, 
+export const API_SUCCESS_MESSAGE_CREATE = `New user with empty password has been created, 
 advice user to reset password as soon as possible.`;
-
+export const API_SUCCESS_MESSAGE_EDIT = 'User has been updated';
 interface Props {
   branding: BrandingType;
   form: FormInstance;
+  user?: DetailedUser
 }
 
-const ModalForm: React.FC<Props> = ({ form, branding }) => (
-  <Form
-    form={form}
-    labelCol={{ span: 8 }}
-    wrapperCol={{ span: 14 }}>
-    <Form.Item
-      label={USER_NAME_LABEL}
-      name={USER_NAME_NAME}
-      required
-      rules={[
-        {
-          message: 'Please type in your username.',
-          required: true,
-        },
-      ]}
-      validateTrigger={[ 'onSubmit' ]}>
-      <Input autoFocus maxLength={128} placeholder="user name" />
-    </Form.Item>
-    <Form.Item
-      label={DISPLAY_NAME_LABEL}
-      name={DISPLAY_NAME_NAME}>
-      <Input maxLength={128} placeholder="display name" />
-    </Form.Item>
-    {branding === BrandingType.Determined ? (
+const ModalForm: React.FC<Props> = ({ form, branding, user }) => {
+  useEffect(() => {
+    form.setFieldsValue({
+      [ADMIN_NAME]: user?.isAdmin,
+      [DISPLAY_NAME_NAME]: user?.displayName,
+    });
+  }, [ user, form ]);
+  return (
+    <Form
+      form={form}
+      labelCol={{ span: 8 }}
+      wrapperCol={{ span: 14 }}>
+      <Form.Item
+        initialValue={user?.username}
+        label={USER_NAME_LABEL}
+        name={USER_NAME_NAME}
+        required
+        rules={[
+          {
+            message: 'Please type in your username.',
+            required: true,
+          },
+        ]}
+        validateTrigger={[ 'onSubmit' ]}>
+        <Input autoFocus disabled={!!user} maxLength={128} placeholder="user name" />
```

Comment:
This component is also used for creating user, when editing, user name is not allowed to be changed.
Are you suggesting we should use `span` instead of `input` for user name when editing?

Reply:
ok i didn't realize it was the same fields used for creating, but that makes sense. i guess i would prefer `span` during editing but that's my personal preference.